Simplenote/src/main/java/com/automattic/simplenote/AddTagActivity.java

```diff
@@ -15,78 +15,60 @@
 import com.automattic.simplenote.widgets.MorphCircleToRectangle;
 import com.automattic.simplenote.widgets.MorphSetup;
 import com.google.android.material.textfield.TextInputEditText;
 import com.google.android.material.textfield.TextInputLayout;
 import com.simperium.client.Bucket;
-import com.simperium.client.BucketObjectNameInvalid;
 
 import java.util.Objects;
 
 public class AddTagActivity extends AppCompatActivity implements TextWatcher {
-    private Bucket<Tag> mBucketTag;
-    private Button mButtonNegative;
-    private Button mButtonPositive;
-    private TextInputEditText mTagInput;
-    private TextInputLayout mTagLayout;
+    private AddTagViewModel viewModel;
+    private ActivityTagAddBinding binding;
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         ThemeUtils.setTheme(this);
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_tag_add);
+        binding = ActivityTagAddBinding.inflate(getLayoutInflater());
 
-        int widthScreen = getResources().getDisplayMetrics().widthPixels;
-        int widthMargin = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 56, getResources().getDisplayMetrics());
-        int widthMaximum = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 312, getResources().getDisplayMetrics());
-        LinearLayout layout = findViewById(R.id.layout);
-        layout.getLayoutParams().width = Math.min(widthMaximum, widthScreen - widthMargin);
-        layout.requestLayout();
-
-        ((View) layout.getParent()).setOnClickListener(
-            new View.OnClickListener() {
-                @Override
-                public void onClick(View view) {
-                    finishAfterTransition();
-                }
-            }
-        );
-        layout.setOnClickListener(null);
-
-        MorphSetup.setSharedElementTransitions(this, layout, getResources().getDimensionPixelSize(R.dimen.corner_radius_dialog));
+        Bucket<Tag> mBucketTag = ((Simplenote) getApplication()).getTagsBucket();
+        ViewModelFactory viewModelFactory = new ViewModelFactory(mBucketTag, this, null);
+        ViewModelProvider viewModelProvider = new ViewModelProvider(this, viewModelFactory);
+        viewModel = viewModelProvider.get(AddTagViewModel.class);
 
-        mBucketTag = ((Simplenote) getApplication()).getTagsBucket();
+        setObservers();
+        setupLayout();
+        setupViews();
 
-        TextView title = findViewById(R.id.title);
-        title.setText(getString(R.string.add_tag));
-
-        mTagLayout = findViewById(R.id.tag_layout);
-        mTagInput = findViewById(R.id.tag_input);
-
-        if (mTagInput != null) {
-            mTagInput.addTextChangedListener(this);
+        setContentView(binding.getRoot());
+    }
 
-            new Handler().postDelayed(
+    private void setupViews() {
+        binding.title.setText(getString(R.string.add_tag));
```

Comment:
NIT: I quite like the something like 
```
private void setupView() = with(binding) {
}
```
If you do that, you don't have to use the `binding.view` calls and just call directly the views. It's little more readable.